app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -139,20 +136,62 @@ class InfectionMessengerRepositoryImpl(
     }
 
     override suspend fun processKeysBasedOnToken(token: String) {
-        if (lastScheduledToken != token) {
-            Timber.e(IllegalArgumentException("processing of token ${token} when we expect to process ${lastScheduledToken}"))
+        val fullSession = sessionDao.getFullSession(token)
+
+        if (fullSession == null || fullSession.session.currentToken != token) {
+            Timber.e(IllegalStateException("Session for token ${token} not found"))
+            return
         }
         val configuration =
             configurationRepository.getConfiguration()
                 ?: throw IllegalStateException("we have no configuration values here, it doesn´t make sense to continue")
 
-        val fullSession: DbFullSession = sessionDao.getFullSession(token)
-
         //TODO: CTAA-1664 cleanup the files and sessionDao (delete by token)
 
-        val currentWarningType = fullSession.session.warningType
+        when (fullSession.session.processingPhase) {
+
+            FullBatch -> {
+                Timber.d("Let´s evaluate based on the summary and WarningType the fullbatch")
+                processResultsOfFullBatch(token, configuration, fullSession)
+            }
+            ProcessingPhase.DailyBatch -> {
+                val exposureInformation = exposureNotificationRepository.getExposureInformationWithPotentiallyInformingTheUser(fullSession.session.currentToken)
+
+                val dates = exposureInformation.extractLatestRedAndYellowContactDate(configuration.dailyRiskThreshold)
+
+                if (dates.firstYellowDay != null && fullSession.session.yellowDay == null){
+                    fullSession.session.yellowDay = dates.firstYellowDay
+                }
 
+                dates.firstRedDay?.let { firstRedDay ->
+                    fullSession.session.yellowDay?.let { yellowDay ->
+                        quarantineRepository.receivedWarning(WarningType.YELLOW, timeOfContact = yellowDay)
+                    }
+
+                    quarantineRepository.receivedWarning(WarningType.RED, timeOfContact = dates.firstRedDay)
+                    return
+                }
+
+                if (fullSession.dailyBatchesParts.isEmpty()){
+                    val yellowDay = fullSession.session.yellowDay
+                    if (yellowDay != null) {
+                        quarantineRepository.receivedWarning(WarningType.YELLOW, timeOfContact = yellowDay)
+                    } else {
+                        Timber.e("we´re done processing but it seems there is no yellow date")
```

Comment:
else-branch can be removed

Reply:
Let's keep the logging for error analysis. It will be dropped in production anyway.
I'll reduce log level to debug though